File: Data Structures & Algorithms/two-integer-sum/submission-11.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}

        for index, v in enumerate(nums):
            print(index, v)
            m[v] = index
        
        for index, v in enumerate(nums):
            t = target - v

            if m.get(t) and m.get(t) != index:
                return [index, m.get(t)]

        return []

File: Data Structures & Algorithms/two-integer-sum-ii/submission-11.ts
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers: number[], target: number): number[] {
        const numMap = {}
        for (let i = 0; i < numbers.length; i++) {
            const n = numbers[i];
            numMap[n] = i;
        }

        for (let i = 0; i < numbers.length; i++) {
            const n = numbers[i];
            if (target - n in numMap && numMap[target - n] !== i) {
                return [i + 1, numMap[target - n] + 1]
            }

        }

        return []
    }
}
